code/modules/mailorder/modr_handling.dm

```diff
@@ -0,0 +1,146 @@
+			radio_controller.get_frequency(FREQ_PDA).post_packet_without_source(pdaSignal)
+
+	proc/deny_order()
+		if(src.orderedby)
+			var/refund_acc = FindBankAccountByName(src.orderedby)
+			if(refund_acc)
+				refund_acc["current_money"] += src.order_cost
+		if(notify_netid)
+			var/datum/signal/pdaSignal = get_free_signal()
+			pdaSignal.data = list("address_1"=notify_netid, "command"="text_message", "sender_name"="CARGO-MAILBOT", "sender"="00000000", "message"="Notification: Mail order denied by local quartermaster service.")
+			radio_controller.get_frequency(FREQ_PDA).post_packet_without_source(pdaSignal)
+
+
+//boxes and loaders for transfer of ordered goods to purchaser
+
+/////////////CONFIGURATION NOTES/////////////
+
+//A mail loading chute + landmark mailorder spawn and target are recommended for map addition.
+//Mail order spawn landmark, located at edge of map
+//Mail order target landmark, placed at location on station that is chute location or will feed box into chute
+//Mail loading chute, located somewhere external, connected to the station's mail loop
+
+//The chute should accept mail from a point external on the station and place it into the mail loop
+//using a juncture that has a tag which doesn't match any mail chute.
+
+//Adding the chute will not prevent purchase of the mail-order via QM secure crate;
+//merely add the option for a less secure but more convenient box-based delivery
+
+//Box for mail-based delivery
+/obj/item/storage/box/mailorder
+	name = "mail-order box"
+	icon_state = "evidence"
+	desc = "A box containing mail-ordered items."
+	var/mail_dest = null //used in mail loop delivery
+
+	proc/yeetself()
+		var/yeetdelay = rand(15 SECONDS,20 SECONDS)
+		SPAWN(yeetdelay)
+			var/yeetbegin = pick_landmark(LANDMARK_MAILORDER_SPAWN)
+			var/yeetend = pick_landmark(LANDMARK_MAILORDER_TARGET)
+			if(!yeetbegin || !yeetend)
+				logTheThing("debug",null,null,"[src] failed to launch at intended destination, tell kubius")
+			src.set_loc(get_turf(yeetbegin))
+			src.throw_at(yeetend, 100, 1)
+
+//Box for QM-based delivery
+/obj/storage/secure/crate/mailorder
+	name = "mail-order crate"
+	desc = "A crate that holds mail-ordered items."
+	personal = 1
+
+	proc/launch_procedure()
+		if(src.registered)
+			src.name = "\improper [src.registered]'s mail-order crate"
+			src.desc = "A crate that holds mail-ordered items. It's registered to [src.registered]'s ID card."
+
+//Auto chute that accepts boxes for mail-based delivery, and nothing else
+/obj/machinery/floorflusher/industrial/mailorder
+	name = "external mail loading chute"
+	desc = "A large chute that only accepts specially designed mail-order boxes."
+	var/destination_tag = null
+
+	Crossed(atom/movable/AM)
+		if(istype(AM,/obj/item/storage/box/mailorder))
+			..()
+
+	MouseDrop_T()
+		return
+
+	attack_hand()
+		return
+
+	attackby(var/obj/item/I, var/mob/user)
+		if(istype(I,/obj/item/storage/box/mailorder))
+			..()
+
+
+	flush()
```

Comment:
This smells like copypasted code. Do you reckon you could split off the part of the parent flush() proc you need to override into a new proc and then override only that instead of overriding the whole thing?